filters/builtin/compress.go

```diff
@@ -140,13 +140,11 @@ func (c *compress) CreateFilter(args []interface{}) (filters.Filter, error) {
 
 	if lf, ok := args[0].(float64); ok && math.Trunc(lf) == lf {
 		f.level = int(lf)
-		_, err := gzip.NewWriterLevel(nil, f.level)
-		if err != nil {
+		if f.level < gzip.HuffmanOnly || f.level > gzip.BestCompression {
```

Comment:
:+1: This is what [NewWriterLevel is doing](https://golang.org/src/compress/gzip/gzip.go?s=1836:1896#L50) internally

Reply:
OTOH filter documentation states:
> The possible values are integers between 0 and 9 (inclusive), where 0 means no-compression, 1 means best-speed and 9 means best-compression.

so the alternative would be to simply use 0 and 9 respectively (i.e. disable undocumented -1 aka DefaultCompression and -2 aka HuffmanOnly)

Reply:
interestingly, it's indeed what the gzip.NewWriterLevel does, but it's not the case with the flate.NewWriter. Traced this far: https://golang.org/src/compress/flate/huffman_bit_writer.go#L93

Reply:
I found this [function](https://golang.org/src/compress/flate/deflate.go#L578) which does the validation of the levels in a `switch` statement. That seemed a bit too elaborate for what the filterspec wants to achieve so I went for the simpler version from the `gzip` package.

I'm a bit confused: Do you want me to change the validation so that the filterspec only accepts 0 - 9 as valid inputs or will this be taken care of in #1517 ?
I can do the change, no problem, just tell me.

Reply:
@wndhydrnt @aryszka Yes, flate uses the aforementioned switch that does the same as far as I can tell (i.e. accepts valid level in range [-2, 9], see default case) 

I hardly can image how users could come up with using -2 or -1 as they are not documented and docs make no connection to go's implementation constants.
So I personally would narrow acceptable values to [0, 9] as documented and close followup #1517.
Do we consider fixing an inconsistency/undocumented behavior a breaking change?